src/com/facebook/buck/jvm/java/intellij/IjModuleFactory.java

```diff
@@ -677,6 +683,42 @@ public void apply(TargetNode<JavaTestDescription.Arg, ?> target, ModuleBuildCont
     }
   }
 
+  private class KotlinLibraryModuleRule implements IjModuleRule<KotlinLibraryDescription.Arg> {
+
+    @Override
+    public Class<? extends Description<?>> getDescriptionClass() {
+      return KotlinLibraryDescription.class;
+    }
+
+    @Override
+    public void apply(
+        TargetNode<KotlinLibraryDescription.Arg, ?> target,
+        ModuleBuildContext context) {
+      addDepsAndSources(
+          target,
+          false /* wantsPackagePrefix */,
```

Comment:
this option is there mostly because fb source layout is weird and doesn't match Intellij convention. At the same time, given the similarity between kotlin and java, it might not be a bad idea to have it. Give it a shot, if it works with this set to `true` maybe just leave it in?